apps/allocations/app/components/App/PeriodDetails.js

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import { useAragonApi } from '../../api-react'
+import { Box, GU, Text, textStyle, useTheme } from '@aragon/ui'
+import throttle from 'lodash.throttle'
+import BigNumber from 'bignumber.js'
+import usePeriod from '../../hooks/usePeriod'
+import { ETHER_TOKEN_FAKE_ADDRESS, isTokenVerified } from '../../../../../shared/utils/token-utils'
+
+const MS_PER_DAY = 24 * 60 * 60 * 1000
+const MONTHS = [ 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec' ]
+const CONVERT_API_BASE = 'https://min-api.cryptocompare.com/data'
+const CONVERT_THROTTLE_TIME = 5000
+
+const convertApiUrl = symbols =>
+  `${CONVERT_API_BASE}/price?fsym=USD&tsyms=${symbols.join(',')}`
+
+const processTime = ms => {
+  const days = Math.round(ms/MS_PER_DAY)
+  if(days%7 === 0){
+    const weeks = days/7
+    return [ String(weeks), `week${weeks === 1 ? '' : 's'}` ]
+  }
+  return [ String(days), `day${days === 1 ? '' : 's'}` ]
+}
+
+const getBudgetTotals = (budgets, rates) => {
+  if(budgets.length && rates){
+    const convertedBudgets = budgets.map(budget => {
+      const rate = rates[budget.token.symbol]
+      return {
+        amount: rate ? (new BigNumber(budget.amount)).div(rate).div(10**budget.token.decimals) : new BigNumber(0),
+        remaining: rate ? (new BigNumber(budget.remaining)).div(rate).div(10**budget.token.decimals) : new BigNumber(0)
+      }
+
+    })
+
+    return convertedBudgets.reduce((total, budget) => {
+      return {
+        amount: total.amount.plus(budget.amount),
+        remaining: total.remaining.plus(budget.remaining)
+      }
+    })
+  } else {
+    return {
+      amount: new BigNumber(0),
+      remaining: new BigNumber(0)
+    }
+  }
+}
+
+const PeriodDetails = () => {
+  const { appState : { budgets = [] } } = useAragonApi()
+  const { startDate, endDate, duration } = usePeriod()
+  const [ rates, setRates ] = useState(null)
+  const { amount, remaining } = getBudgetTotals(budgets, rates)
+  const durationArray = processTime(duration)
+  const remainingArray = processTime(endDate - new Date())
+
+  const updateConvertedRates = throttle(async () => {
+    if(budgets.length) {
+      const symbols = budgets.filter(({ token }) => isTokenVerified(token.symbol))
+        .map(({ token }) => token.symbol)
+      const res = await fetch(convertApiUrl(symbols))
+      const convertRates = await res.json()
+      if (JSON.stringify(rates) !== JSON.stringify(convertRates)) {
+        setRates(convertRates)
+      }
+    }
+  }, CONVERT_THROTTLE_TIME)
+
+  updateConvertedRates()
```

Comment:
On every render of this component, `updateConvertedRates` will get re-defined and re-called. I don't think we want that. Instead, we can make use of the `useEffect` hook like this:

```js
const [ ratesFetchedAt, setRatesFetchedAt ] = useState(new Date())

useEffect(() => {
  const interval = setInterval(() => setRatesFetchedAt(new Date()), CONVERT_INTERVAL)
  return () => clearInterval(interval)
}, [])

useEffect(() => {
  async function updateConvertedRates() {
    ...
  }
  updateConvertedRates()
}, [ ratesFetchedAt ])
```

The second argument to the first `useEffect` here is `[]` – this means that the effect will be run _once_, when the component first renders, and not again. The return value of the same effect, `() => clearInterval(interval)`, will be run when the component unmounts.

The second effect will be run when the component first mounts and then every time the `ratesFetchedAt` variable changes.